internal/server/server.go

```diff
@@ -0,0 +1,135 @@
+// SPDX-FileCopyrightText: 2025 The Kepler Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package server
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/exporter-toolkit/web"
+	"github.com/sustainable-computing-io/kepler/internal/service"
+)
+
+// APIService defines the interface for the HTTP server providing API endpoints
+type APIService interface {
+	service.Service
+	Register(endpoint, summary, description string, handler http.Handler) error
+}
+
+// APIServer implements APIServer
+type APIServer struct {
+	// input
+	logger      *slog.Logger
+	listenAddrs []string
+
+	// http
+	server              *http.Server
+	mux                 *http.ServeMux
+	endpointDescription string
```

Comment:
question: is this only used for testing?

Reply:
no, actually, I removed the part that builds the landing page. I am adding that now.